crengine/src/private/lvfreetypeface.cpp

```diff
@@ -1801,25 +1820,30 @@ LVFontGlyphCacheItem *LVFreeTypeFace::getGlyph(lUInt32 ch, lChar32 def_char, lUI
                 rend_flags );             /* load flags, see below */
         if ( error == FT_Err_Execution_Too_Long && _hintingMode == HINTING_MODE_BYTECODE_INTERPRETOR ) {
             // Native hinting bytecode may fail with some bad fonts: try again with no hinting
+            CRLog::error("Font '%s': loading glyph too long!", _fileName.c_str());
             rend_flags |= FT_LOAD_NO_HINTING;
             error = FT_Load_Glyph( _face, ch_glyph_index, rend_flags );
         }
         if ( error ) {
             return NULL;  /* ignore errors */
         }
 
-        if (_embolden) {
-            FT_GlyphSlot_Embolden(_slot); // See setEmbolden() for details
+        if (_synth_weight > 0) {
+            if ( _slot->format == FT_GLYPH_FORMAT_OUTLINE ) {
+                // See setSynthWeight() for details
+                FT_Outline_Embolden(&_slot->outline, 2 * _synth_weight_half_strength);
+                FT_Outline_Translate(&_slot->outline, -_synth_weight_half_strength, -_synth_weight_half_strength);
+            }
         }
         if (_italic == 2) {
             FT_GlyphSlot_Oblique(_slot);
         }
-        if (_embolden || _italic==2) {
+        if (_synth_weight > 0 || _italic == 2) {
             // Render now that transformations are applied
             FT_Render_Glyph(_slot, _drawMonochrome?FT_RENDER_MODE_MONO:FT_RENDER_MODE_LIGHT);
```

Comment:
Can't test that more, but regarding my issue with italic sometimes not working, no issue for you with embolden+italic ?
It might feel that FT_GlyphSlot_Embolden + FT_GlyphSlot_Oblique can work together - but FT_Outline_Embolden+FT_Outline_Translate + FT_GlyphSlot_Oblique might not ? or needs re-ordering?

Although we already have it like that below in getGlyphByIndex() :/ so it must work.

Reply:
As you know `FT_GlyphSlot_Embolden` no more relevant, so I use `FT_Outline_Embolden`.
Yes it work with embolden+italic.
![Screenshot_20210331_221815](https://user-images.githubusercontent.com/36960933/113191816-0f42e700-926f-11eb-973b-e1e411154a02.png)